src/rpc/blockchain.cpp

```diff
@@ -214,20 +214,9 @@ UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex
 
 UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIndex* blockindex, llmq::CChainLocksHandler& clhandler, llmq::CInstantSendManager& isman, bool txDetails)
 {
-    // Serialize passed information without accessing chain state of the active chain!
-    AssertLockNotHeld(cs_main); // For performance reasons
+    UniValue result = blockheaderToJSON(tip, blockindex, clhandler, isman);
```

Comment:
21114: this PR changes the ordering, should adjust `getblock` help text accordingly imo

Reply:
@thephez thoughts? 

IMO it doesn't matter since the JSON RPC API isn't order-sensitive and changing the order would make backports mildly unpleasant.

Reply:
The output in rpc console/dash-cli will be different and we should preserve the same order in help text imo. Not sure why they simply ignored this in bitcoin...

Reply:
That's a Map/Dictionary, order of members is not important accordingly to JSON spec...
Anyone who is expecting to see specific order - should not do that.
Moreover, many implementation of json do break order of Map/dictionary when input/output. Implementation that do not changes order are mutants IMO

Reply:
All our rpc commands return fields in the same order they are listed in rpc help text. Don't see a reason why we should break it.

Reply:
Not sure why I didn't see the ping, but just opened the PR and noticed it :man_shrugging: 

>All our rpc commands return fields in the same order they are listed in rpc help text. Don't see a reason why we should break it.

I concur with this. I rely on that consistency when updating docs and it would be annoying to lose that.

Reply:
It does ensure that whatever is reported in `getblockheader` is present in `getblock` by deduplicating logic, I think there is some value to it. I've updated the commit to rearrange the help message to match the order.